plugins/i3s/io/EsriReader.cpp

```diff
@@ -0,0 +1,435 @@
+        const std::string geomUrl = localUrl + "/geometries/";
+        auto xyz = fetchBinary(geomUrl, "0", ".bin.pccxyz");
+        std::vector<lepcc::Point3D> pointcloud = decompressXYZ(&xyz);
+
+        std::vector<int> selected;
+        uint64_t startId;
+
+        {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            startId = view->size();
+
+            for (uint64_t j = 0; j < pointcloud.size(); ++j)
+            {
+                double x = pointcloud[j].x;
+                double y = pointcloud[j].y;
+                double z = pointcloud[j].z;
+
+                if (m_bounds.contains(x, y, z))
+                {
+                    PointId id = view->size();
+                    selected.push_back(j);
+                    view->setField(pdal::Dimension::Id::X, id, pointcloud[j].x);
+                    view->setField(pdal::Dimension::Id::Y, id, pointcloud[j].y);
+                    view->setField(pdal::Dimension::Id::Z, id, pointcloud[j].z);
+                }
+            }
+        }
+
+        const std::string attrUrl = localUrl + "/attributes/";
+
+        for (const auto& dimEntry : m_dimMap)
+        {
+            const Dimension::Id dimId(dimEntry.first);
+            const uint64_t key(dimEntry.second.key);
+            std::string dataType(dimEntry.second.dataType);
+
+            if (dimId == Dimension::Id::Red)
+            {
+                auto data = fetchBinary(
+                        attrUrl, std::to_string(key), ".bin.pccrgb");
+                std::vector<lepcc::RGB_t> rgbPoints = decompressRGB(&data);
+
+                std::lock_guard<std::mutex> lock(m_mutex);
+
+                for (std::size_t i(0); i < selected.size(); ++i)
+                {
+                    view->setField(pdal::Dimension::Id::Red,
+                            startId + i, rgbPoints[selected[i]].r);
+                    view->setField(pdal::Dimension::Id::Green,
+                            startId + i, rgbPoints[selected[i]].g);
+                    view->setField(pdal::Dimension::Id::Blue,
+                            startId + i, rgbPoints[selected[i]].b);
+                }
+            }
+            else if (dimId == Dimension::Id::Intensity)
+            {
+                auto data = fetchBinary(attrUrl, std::to_string(key),
+                        ".bin.pccint");
+
+                std::vector<uint16_t> intensity = decompressIntensity(&data);
+
+                std::lock_guard<std::mutex> lock(m_mutex);
+                for (std::size_t i(0); i < selected.size(); ++i)
+                {
+                    view->setField(pdal::Dimension::Id::Intensity,
+                            startId + i, intensity[selected[i]]);
+                }
+            }
+            else
+            {
+                const auto data = fetchBinary(
+                        attrUrl, std::to_string(key), ".bin.gz");
+
+                std::lock_guard<std::mutex> lock(m_mutex);
+
+                // TODO Correct types.
```

Comment:
Address this TODO.  Presumably the solution here is the floating point types which seem ambiguous here.  Is there a list of these types in the spec somewhere?